src/stub/src/arch/powerpc/64le/bxx.S

```diff
@@ -37,7 +37,7 @@ ppcbxx:  # (*f_unf)(xo->buf, out_len, h.b_cto8, h.b_ftid);
 #define ptr0 a4
 
         cmplwi cr0,ftid,0xd0;       bnelr- cr0  # if (0xd0!=ftid)   return;
-        rlwinm. len,len,32-2,2,31; beqlr- cr0  # if (0==(len>>=2)) return;
+        sradi len,len,2; beqlr- cr0  # if (0==(len>>=2)) return;
```

Comment:
NO.  You MUST use  "sradi."  in order to record the condition.  Also, this is nearly cheating because the shift is Arithmetic (sign extending: ~0 (==-1) is a fixed point instead of becoming 0x3ffff... . The reason 'sradi.' MIGHT be acceptable is that a length >= 2\*\*63 is totally unreasonable.  [In fact, the compression side currently has a maximum uncompressed length of 0xffffffff (2\*\*32 -1).]